griptape/config/drivers/base_driver_config.py

```diff
@@ -22,15 +23,57 @@
 
 @define
 class BaseDriverConfig(ABC, SerializableMixin):
-    prompt: BasePromptDriver = field(kw_only=True, metadata={"serializable": True})
-    image_generation: BaseImageGenerationDriver = field(kw_only=True, metadata={"serializable": True})
-    image_query: BaseImageQueryDriver = field(kw_only=True, metadata={"serializable": True})
-    embedding: BaseEmbeddingDriver = field(kw_only=True, metadata={"serializable": True})
-    vector_store: BaseVectorStoreDriver = field(kw_only=True, metadata={"serializable": True})
-    conversation_memory: Optional[BaseConversationMemoryDriver] = field(
-        default=None,
-        kw_only=True,
-        metadata={"serializable": True},
-    )
-    text_to_speech: BaseTextToSpeechDriver = field(kw_only=True, metadata={"serializable": True})
-    audio_transcription: BaseAudioTranscriptionDriver = field(kw_only=True, metadata={"serializable": True})
+    _prompt: BasePromptDriver = field(kw_only=True, default=None, metadata={"serializable": True}, alias="prompt")
+    _image_generation: BaseImageGenerationDriver = field(
+        kw_only=True, default=None, metadata={"serializable": True}, alias="image_generation"
+    )
+    _image_query: BaseImageQueryDriver = field(
+        kw_only=True, default=None, metadata={"serializable": True}, alias="image_query"
+    )
+    _embedding: BaseEmbeddingDriver = field(
+        kw_only=True, default=None, metadata={"serializable": True}, alias="embedding"
+    )
+    _vector_store: BaseVectorStoreDriver = field(
+        default=None, kw_only=True, metadata={"serializable": True}, alias="vector_store"
+    )
+    _conversation_memory: Optional[BaseConversationMemoryDriver] = field(
+        default=None, kw_only=True, metadata={"serializable": True}, alias="conversation_memory"
+    )
+    _text_to_speech: BaseTextToSpeechDriver = field(
+        default=None, kw_only=True, metadata={"serializable": True}, alias="text_to_speech"
+    )
+    _audio_transcription: BaseAudioTranscriptionDriver = field(
+        default=None, kw_only=True, metadata={"serializable": True}, alias="audio_transcription"
+    )
+
+    @lazy_property()
```

Comment:
Is this required on the abstract method? (Like is it functionally required vs just for documentation? What happens if a subclass forgets it?)

Reply:
It's not technically required, but I think it's good for documentation. If a subclass forgets it, everything should still work fine but that field will not be lazy loaded.